File: frontend/src/types/subscription.ts
export type SubscriptionStatus = 'Active' | 'Cancelled' | 'Paused'

/** Espelha o JSON retornado por GET /api/subscriptions/{id} e /api/subscriptions/by-customer/{id}. */
export interface Subscription {
  id: string
  customerId: string
  planId: string
  planName: string
  planPrice: number
  startDate: string
  nextBillingDate: string
  endDate: string | null
  status: SubscriptionStatus
}

File: src/SmartRecovery.Domain/Entities/Subscription.cs
using SmartRecovery.Domain.Enums;

namespace SmartRecovery.Domain.Entities;

public class Subscription : BaseEntity
{
    public Guid CustomerId { get; set; }
    public Guid PlanId { get; set; }

    public DateTime StartDate { get; set; }
    public DateTime NextBillingDate { get; set; }
    public DateTime? EndDate { get; set; }
    public SubscriptionStatus Status { get; set; } = SubscriptionStatus.Active;

    public Customer Customer { get; set; } = null!;
    public Plan Plan { get; set; } = null!;
    public ICollection<Payment> Payments { get; set; } = [];
}
